dijkstra.c

```diff
@@ -0,0 +1,125 @@
+#include "heap.h"
+#include "dijkstra.h"
+#include "characters.h"
+#include <stdio.h>
+
+
+//compares cost as apposed to sheaffers height map compare
+static int32_t cost_cmp(const void *key, const void *with) {
+  return ((struct character_s *)key)->cost - ((struct character_s * )with)-> cost;
+}
+
+// static int32_t edge_penalty(int8_t x, int8_t y)
+// {
+//   return (x == 1 || y == 1 || x == GRID_WIDTH - 2 || y == GRID_HEIGHT - 2) ? 2 : 1;
+// }
+
+// static int32_t isValid(int8_t x, int8_t y) {
+//     return x >= 0 && x < GRID_WIDTH && y >= 0 && y < GRID_HEIGHT;
+// }
+
+void dijkstra_map(terrain_map_t *map, Location_t *start, dijk_map_t *dNode[GRID_HEIGHT][GRID_WIDTH], dijk_map_t *dn, struct character_s *npc)
```

Comment:
you asked about dn, isn't dNode and dn supposed to be the same thing fundamentally? you only need 1 array to collapse into what the actual grid looks like. Again if is supposed to be a return parameter you shouldn't need the array sizes known in the parameter? and just treat it like an array pointer

Reply:
my idea with dn is to be a temp node, but i think im making it way more coplicated than it should be.

Reply:
there is no such thing as a temp node?